### PR TITLE
Expose some apis

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

correspond https://github.com/rizinorg/cutter/pull/2948

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

File: librz/core/cfile.c [hunk 1]
```diff
@@ -1600,7 +1603,13 @@ RZ_IPI void rz_core_io_file_open(RzCore *core, int fd) {
...
 
-RZ_IPI void rz_core_io_file_reopen(RzCore *core, int fd, int perms) {
+/**
+ * \brief Reopen file
+ * \param core RzCore instance
+ * \param fd File descriptor
+ * \param perms Permission \s RZ_PERM_R
+ */
+RZ_API void rz_core_io_file_reopen(RzCore *core, int fd, int perms) {
 	if (rz_io_reopen(core->io, fd, perms, 644)) {
```
Comment (wargio) [inline librz/core/cfile.c]: RZ_NONNULL and add a rz_return_if for core and fd

File: librz/core/cfile.c [hunk 2]
```diff
@@ -1556,9 +1556,12 @@ RZ_API RzCoreFile *rz_core_file_cur(RzCore *r) {
...
 
-/* --------------------------------------------------------------------------------- */
-
-RZ_IPI void rz_core_io_file_open(RzCore *core, int fd) {
+/**
+ * \brief Open file use read-only Permission
+ * \param core RzCore instance
+ * \param fd File descriptor
+ */
+RZ_API void rz_core_io_file_open(RzCore *core, int fd) {
```
Comment (wargio) [inline librz/core/cfile.c]: same check here

File: librz/core/cdebug.c [hunk 3]
```diff
@@ -227,12 +227,17 @@ RZ_IPI void rz_core_debug_single_step_over(RzCore *core) {
 	rz_config_set_b(core->config, "io.cache", io_cache);
 }
 
-RZ_IPI void rz_core_debug_breakpoint_toggle(RzCore *core, ut64 addr) {
+/**
+ * \brief Toggle breakpoint
+ * \param core RzCore instance
+ * \param addr Breakpoint addr
+ */
+RZ_API void rz_core_debug_breakpoint_toggle(RzCore *core, ut64 addr) {
```
Comment (wargio) [inline librz/core/cdebug.c]: same here

File: librz/core/cil.c [hunk 4]
```diff
@@ -47,7 +47,11 @@ RZ_IPI void rz_core_analysis_esil_init(RzCore *core) {
...
 }
 
-RZ_IPI void rz_core_analysis_esil_reinit(RzCore *core) {
+/**
+ * \brief Reinitialize ESIL
+ * \param core RzCore reference
+ */
+RZ_API void rz_core_analysis_esil_reinit(RzCore *core) {
 	rz_analysis_esil_free(core->analysis->esil);
 	core_esil_init(core);
```
Comment (wargio) [inline librz/core/cil.c]: same here

File: librz/core/cil.c [hunk 5]
```diff
@@ -127,7 +131,7 @@ static char *get_esil_stack_name(RzCore *core, const char *name, ut64 *addr, ut3
  * \param addr Base address of the stack region, if UT64_MAX it is automatically computed
  * \param size Size of the stack region, if UT32_MAX it is automatically computed
  */
-RZ_IPI void rz_core_analysis_esil_init_mem(RzCore *core, const char *name, ut64 addr, ut32 size) {
+RZ_API void rz_core_analysis_esil_init_mem(RzCore *core, RZ_NULLABLE const char *name, ut64 addr, ut32 size) {
 	ut64 current_offset = core->offset;
```
Comment (wargio) [inline librz/core/cil.c]: add checks and rz_nonnull

**Review by ret2libc — COMMENTED** [review]
What is the reason for exposing these APIs? Could you add a min description to the PR? Also, why just few of them and not, for example, `rz_core_analysis_esil_init`, `rz_core_analysis_esil_step_over_until`, `rz_core_debug_continue`, etc.?

**Comment by b1llow** [comment]
> What is the reason for exposing these APIs? Could you add a min description to the PR? Also, why just few of them and not, for example, `rz_core_analysis_esil_init`, `rz_core_analysis_esil_step_over_until`, `rz_core_debug_continue`, etc.?

Just because of the needs of Cutter, and then I just exposed the minimum required APIs.
Then this PR is the premise of another PR

https://github.com/rizinorg/cutter/pull/2948

File: librz/core/cil.c (hunk 5 above)
Comment (b1llow) [reply librz/core/cil.c]: https://github.com/rizinorg/rizin/blob/2b9e68233f1cc0d6c12dcdc50f3a982d2a7173e7/librz/core/cil.c#L129-L130
https://github.com/rizinorg/rizin/blob/2b9e68233f1cc0d6c12dcdc50f3a982d2a7173e7/librz/core/cmd/cmd_analysis.c#L1879-L1880

it should be nullable

File: librz/core/cfile.c (hunk 1 above)
Comment (b1llow) [reply librz/core/cfile.c]: `fd` is not a pointer.  So maybe it shouldn't be check in `rz_return_if`

File: librz/core/cil.c (hunk 5 above)
Comment (wargio) [reply librz/core/cil.c]: no, i'm talking about RzCore, not name.

File: librz/core/cfile.c (hunk 1 above)
Comment (wargio) [reply librz/core/cfile.c]: fd cannot be negative

File: librz/core/cil.c (hunk 5 above)
Comment (b1llow) [reply librz/core/cil.c]: OK

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**